src/communication/vchannel.cpp

```diff
@@ -77,12 +83,18 @@ aos::Error VChannel::Read(aos::Array<uint8_t>& data, size_t size)
 
 aos::Error VChannel::Write(const aos::Array<uint8_t>& data)
 {
-    auto ret = vch_write(&mWriteHandle, data.Get(), data.Size());
-    if (ret < 0) {
-        return AOS_ERROR_WRAP(ret);
+    size_t written = 0;
+
+    while (written < data.Size()) {
+        auto ret = vch_write(&mWriteHandle, data.Get() + written, data.Size() - written);
+        if (ret < 0) {
+            return AOS_ERROR_WRAP(ret);
+        }
+
+        written += ret;
     }
 
-    if (static_cast<size_t>(ret) != data.Size()) {
+    if (written != data.Size()) {
```

Comment:
that's mostly a check for implementation error: written data should not exceed size.
in this case assert is better.

Reply:
It could exceeds runtime.

Reply:
ok